src/System Application/App/Azure AD Plan/src/User Details/PlanUserDetails.Codeunit.al

```diff
@@ -0,0 +1,43 @@
+// ------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// ------------------------------------------------------------------------------------------------
+
+namespace System.Azure.Identity;
+
+using System.Security.User;
+
+codeunit 776 "Plan User Details"
+{
+    Access = Internal;
+    InherentEntitlements = X;
+    InherentPermissions = X;
+
+    [EventSubscriber(ObjectType::Codeunit, Codeunit::"User Details", 'OnAddUserDetails', '', false, false)]
+    local procedure AddPlanUserDetails(UserSecId: Guid; var UserDetails: Record "User Details")
+    var
+        PlanIds: Codeunit "Plan Ids";
+        AzureADPlan: Codeunit "Azure AD Plan";
+        UsersInPlans: Query "Users in Plans";
+        UserPlansTextBuilder: TextBuilder;
+    begin
+        UsersInPlans.SetRange(User_Security_ID, UserSecId);
+        if UsersInPlans.Open() then
+            while UsersInPlans.Read() do begin
+                UserPlansTextBuilder.Append(UsersInPlans.Plan_Name);
+                UserPlansTextBuilder.Append(' ; ');
+            end;
+
+        UserDetails."User Plans" := CopyStr(UserPlansTextBuilder.ToText().TrimEnd(' ; '), 1, MaxStrLen(UserDetails."User Plans"));
+        UserDetails."Is Delegated" := AzureADPlan.IsPlanAssignedToUser(PlanIds.GetDelegatedAdminPlanId(), UserSecID) or AzureADPlan.IsPlanAssignedToUser(PlanIds.GetHelpDeskPlanId(), UserSecID) or AzureADPlan.IsPlanAssignedToUser(PlanIds.GetD365AdminPartnerPlanId(), UserSecID);
```

Comment:
Could you multiline this and align spacing such that it's easy to see the difference between each call (the PlanIds reference). That would make this much more readable.